components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -141,6 +141,39 @@ def fileread_gen(fin, fsize, bufsize):
     fin.close()
 
 
+def countAnnotations(queryResult=[]):
+    """
+    Count the different the annotions from the
+    provided projection query result
+    """
+
+    counts = {
+        "TagAnnotation": 0,
+        "FileAnnotation": 0,
+        "CommentAnnotation": 0,
+        "LongAnnotation": 0,
+        "MapAnnotation": 0,
+        "OtherAnnotation": 0}
+
+    for r in queryResult:
```

Comment:
It doesn't make much of a difference if I get `/basic/num/long/` or `ome.model.annotations.LongAnnotation`, I still have to loop over the result, and check if the LongAnnotation is only a LongAnnotation or a rating, as well as aggregate XML, Boolean, whatever Annotation to "OtherAnnotation".